scripts/googleAppScripts/SheetToGithubIssue.gs

```diff
@@ -1,100 +1,177 @@
-  };
-
-  try {
-    var response = UrlFetchApp.fetch(url, options);
-    Logger.log(response.getContentText());
-  } catch (error) {
-    Logger.log('Error creating GitHub issue: ' + error.toString());
-  }
+    var personalGithubToken = PropertiesService.getUserProperties().getProperty('GITHUB_TOKEN');
+    var repositoryName = PropertiesService.getUserProperties().getProperty('GITHUB_REPOSITORY');
+    var url = 'https://api.github.com/repos/' + repositoryName + '/issues';
+
+    var payload = {
+        title: title, body: body, labels: ['new-initiative-request']
+    };
+
+    var options = {
+        method: 'post', headers: {
+            Authorization: 'Bearer ' + personalGithubToken, Accept: 'application/vnd.github+json'
+        }, contentType: 'application/json', payload: JSON.stringify(payload),
+    };
+
+    try {
+        var response = UrlFetchApp.fetch(url, options);
+        console.log(response.getContentText());
+    } catch (error) {
+        console.error('Error creating GitHub issue: ' + error.toString());
+        throw error;
+    }
 }
 
 function testOnFormSubmit(row = 153) {
-  var sheet = SpreadsheetApp.getActiveSpreadsheet().getActiveSheet();
-  // Make sure this range includes all the headers for your form responses
-  var headersRange = sheet.getRange("A1:Z1");
-  var headers = headersRange.getValues()[0];
+    var sheet = SpreadsheetApp.getActiveSpreadsheet().getActiveSheet();
+    var headersRange = sheet.getRange("A1:Z1");
+    var headers = headersRange.getValues()[0];
+
+    // Adjust this range to match the range where your test data is located
+    var testDataRange = sheet.getRange(`A${row}:Z${row}`);
+    var testData = testDataRange.getValues()[0];
 
-  // Adjust this range to match the range where your test data is located
-  var testDataRange = sheet.getRange(`A${row}:Z${row}`);
-  var testData = testDataRange.getValues()[0];
+    // Log the actual values to make sure they're what you expect
+    Logger.log("Headers: " + headers.join(", "));
+    Logger.log("Test Data: " + testData.join(", "));
 
-  // Log the actual values to make sure they're what you expect
-  Logger.log("Headers: " + headers.join(", "));
-  Logger.log("Test Data: " + testData.join(", "));
+    var namedValues = {};
+    for (var i = 0; i < headers.length; i++) {
+        namedValues[headers[i]] = [testData[i]];
+    }
 
-  var namedValues = {};
-  for (var i = 0; i < headers.length; i++) {
-    namedValues[headers[i]] = [testData[i]];
-  }
+    Logger.log("Named Values: " + JSON.stringify(namedValues));
 
-  Logger.log("Named Values: " + JSON.stringify(namedValues));
+    var e = {namedValues: namedValues};
+    onFormSubmit(e);
+}
+
+function setGithubCredentials() {
+  setPersonalGithubToken();
+  setGithubRepositoryName();
+}
 
-  var e = { namedValues: namedValues };
-  onFormSubmit(e);
+function setPersonalGithubToken() {
+  var personalGithubToken = 'YOUR_ACTUAL_GITHUB_TOKEN'; // Replace with your actual token
```

Comment:
The idea was to take this from env var (`process.env.GITHUB_TOKEN`), that way the file wouldn't have to be modified (and recorded in the git file index, accidentally committed, etc). Same comment for repo name below.

Reply:
But how will it work? 
This script is not working from the git repo; its here for the documentation purposes. We are running it from Google Apps scripts. How will this integration work?

Reply:
In that case, the value should be taken from the user properties (`PropertiesService.getUserProperties().getProperty('GITHUB_TOKEN')`), so each user could use the same script and only update their GAS env